src/util/Optional.h

```diff
@@ -102,6 +102,19 @@ class Optional {
         return !operator==(other);
     }
 
+    T* operator->()
+    {
+        ASSERT(hasValue());
+        return &m_value;
+    }
+
+    const T* operator->() const
+    {
+        ASSERT(hasValue());
+        return m_value;
```

Comment:
It seems that this function needs to return the address of `m_value`, not the value itself because `m_value` is not a pointer type of `T`.

Reply:
Opps!!

Reply:
I've updated my pr